x-pack/protocol/src/main/java/org/elasticsearch/protocol/xpack/ml/datafeed/DatafeedConfig.java

```diff
@@ -0,0 +1,347 @@
+
+    public static final int DEFAULT_SCROLL_SIZE = 1000;
+
+    // Used for QueryPage
+    public static final ParseField RESULTS_FIELD = new ParseField("datafeeds");
+
+    public static final ParseField ID = new ParseField("datafeed_id");
+    public static final ParseField QUERY_DELAY = new ParseField("query_delay");
+    public static final ParseField FREQUENCY = new ParseField("frequency");
+    public static final ParseField INDEXES = new ParseField("indexes");
+    public static final ParseField INDICES = new ParseField("indices");
+    public static final ParseField TYPES = new ParseField("types");
+    public static final ParseField QUERY = new ParseField("query");
+    public static final ParseField SCROLL_SIZE = new ParseField("scroll_size");
+    public static final ParseField AGGREGATIONS = new ParseField("aggregations");
+    public static final ParseField AGGS = new ParseField("aggs");
+    public static final ParseField SCRIPT_FIELDS = new ParseField("script_fields");
+    public static final ParseField SOURCE = new ParseField("_source");
+    public static final ParseField CHUNKING_CONFIG = new ParseField("chunking_config");
+
+    public static final ObjectParser<Builder, Void> PARSER = new ObjectParser<>("datafeed_config", true, Builder::new);
+
+    static {
+        PARSER.declareString(Builder::setId, ID);
+        PARSER.declareString(Builder::setJobId, Datafeed.JOB_ID);
+        PARSER.declareStringArray(Builder::setIndices, INDEXES);
+        PARSER.declareStringArray(Builder::setIndices, INDICES);
+        PARSER.declareStringArray(Builder::setTypes, TYPES);
+        PARSER.declareString((builder, val) ->
+            builder.setQueryDelay(TimeValue.parseTimeValue(val, QUERY_DELAY.getPreferredName())), QUERY_DELAY);
+        PARSER.declareString((builder, val) ->
+            builder.setFrequency(TimeValue.parseTimeValue(val, FREQUENCY.getPreferredName())), FREQUENCY);
+        PARSER.declareObject(Builder::setQuery, (p, c) -> AbstractQueryBuilder.parseInnerQueryBuilder(p), QUERY);
+        PARSER.declareObject(Builder::setAggregations, (p, c) -> AggregatorFactories.parseAggregators(p), AGGREGATIONS);
+        PARSER.declareObject(Builder::setAggregations, (p, c) -> AggregatorFactories.parseAggregators(p), AGGS);
+        PARSER.declareObject(Builder::setScriptFields, (p, c) -> {
+            List<SearchSourceBuilder.ScriptField> parsedScriptFields = new ArrayList<>();
+            while (p.nextToken() != XContentParser.Token.END_OBJECT) {
+                parsedScriptFields.add(new SearchSourceBuilder.ScriptField(p));
+            }
+            parsedScriptFields.sort(Comparator.comparing(SearchSourceBuilder.ScriptField::fieldName));
+            return parsedScriptFields;
+        }, SCRIPT_FIELDS);
+        PARSER.declareInt(Builder::setScrollSize, SCROLL_SIZE);
+        // TODO this is to read former _source field. Remove in v7.0.0
+        PARSER.declareBoolean((builder, value) -> {
```

Comment:
Again, unsure about removing this field. It should not be necessary (according to the comment) in `v7.0.0`.

Reply:
Since the parser is lenient it will handle any `_source` field by ignoring it anyway, so there's no need to explicitly ignore it and the line can be deleted here.  (This line was needed for the strict parser because it would have errored on finding an `_source` field otherwise.)

As for 7.0 TODOs in general, we have an issue open to go through and action them, so maybe that's something you could do in a separate PR later this week.